### PR TITLE
unregister-error-cleanup

### PR DESCRIPTION
**Pull request by jonohodgson**
- ensure deleteEmptyObjects.except array is being used in the deletePath util
- add conditional nested object to Form demo
- fix broken Dropdown test

This is to fix an issue where nested error objects were not being cleared correctly when a field with a nested object reference was unregistered.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1 +0,0 @@
-npm run prettier && npm run lint && npm test
```
Comment (moroshko) [inline .husky/pre-commit:1]: Why delete the `.husky` files?

File: src/utils/objectPath.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ export function deletePath(obj, path, { deleteEmptyObjects = false } = {}) {
     const shouldDeleteKey =
       deleteEmptyObjects === true ||
       (Array.isArray(deleteEmptyObjects) &&
```
Comment (moroshko) [inline src/utils/objectPath.js]: Should it be `Array.isArray(deleteEmptyObjects.except)`?

File: website/src/pages/components/form/index.js [hunk 3]
```diff
@@ -155,20 +160,25 @@ function FormPage() {
...
-              <Checkbox label="Do you like ice cream?" hideLabel name="likeIceCream" helpText="You MUST like it!">I like ice cream</Checkbox>
-              <RadioGroup name="hungry" label="Are you hungry?" options={hungryOptions} />
-              <Frequency name="salary" label="Salary (before tax)" amountPrefix="$" />
-              <DatePicker name="birthDate" label="Birth date" />
-              <TimeSpan name="age" label="Age" />
-              <Textarea name="aboutYourself" label="Tell us about yourself" height="100" />
-              <Button type="submit">Submit</Button>
-            </Stack>
+             {(formData) => (
+              <Stack gap="8">
```
Comment (moroshko) [inline website/src/pages/components/form/index.js:164]: doesn't seem to be formatted. can you run `npm run prettier`?

File: src/utils/objectPath.js (hunk 2 above)
Comment (jonohodgson) [reply src/utils/objectPath.js]: yes, had deleted testing something and forgot to add back

File: website/src/pages/components/form/index.js (hunk 3 above)
Comment (jonohodgson) [reply website/src/pages/components/form/index.js:164]: ran prettier.  has no effect

**Review by moroshko — DISMISSED** [review]

**Review by moroshko — APPROVED** [review]

**Pull request merged**